### PR TITLE
README

### PR DESCRIPTION
**Pull request by RStankov**
Time to have proper README and CHANGELOG for the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+});
+```
+
+#### Usage
+
+```js
+import api from 'ph/api';
+
+const { body } = api.about()                      // GET /about
+const { body } = api.posts.findOne({ id: 1 })     // GET /posts/1
```
Comment (ayrton) [inline README.md]:
```
const { body } = api.posts.findOne({ id: 1 })     // GET /posts/1
```

This technically does a `GET /posts/1?id=1` right now (even though that's a bug)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+```js
+export default duxy(options, ({ get }) => {
+  get('root', { path: '/'});
+  get('search');
+});
+```
+
+```js
+api.root()                  // GET /
+api.search({ query: 'foo'}) // GET /search?query=foo
```
Comment (ayrton) [inline README.md]: `api.search({ query: 'foo' })`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+const { body } = api.posts.findOne({ id: 1 })     // GET /posts/1
+const { body } = api.posts.findAll({ limit: 10 }) // GET /posts?limit=10
+```
+#### DSL
+
+##### `get(name, { path })`
+
+```js
+export default duxy(options, ({ get }) => {
+  get('root', { path: '/'});
```
Comment (ayrton) [inline README.md]:
``` js
get('root', { path: '/' });
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+##### `post(name, { path })`
+
+```js
+export default duxy(options, ({ post }) => {
+  post('create')
+  post('createPost', { path: '/posts' });
+});
+```
+
+```js
```
Comment (ayrton) [inline README.md:141]: missing `api.create` example

File: README.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+export default duxy(options, ({ get }) => {
+  get('root', { path: '/'});
+  get('search');
+});
+```
+
+```js
+api.root()                  // GET /
```
Comment (ayrton) [inline README.md]: missing `;` in some examples - imo let's keep same style throughout the readme (with or without)

File: README.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+```js
+api.users.findAll({ id: 1 });        // GET /users
+api.users.findOne({ id: 1 });        // GET /users/1
+api.users.create({ id: 1, name });   // POST /users/1
+api.users.update({ id: 1, name });   // PUT /users/1
+api.users.delete({ id: 1 });         // DELETE /users/1
+api.users.moreInfo({ userId: 1 });   // GET /users/1/info
+```
+
+##### `resource(name, { path, only: ['findOne', 'create', 'update', 'delete'] }, fn)`
```
Comment (andreasklinger) [inline README.md]: put this first and recommend it as default usage

**Review by ayrton — APPROVED** [review]
Looks very good

File: README.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
+# Duxy
+
+[![npm version](https://img.shields.io/npm/v/duxy.svg)](https://www.npmjs.com/package/duxy) [![License](https://img.shields.io/npm/l/duxy.svg)](https://www.npmjs.com/package/duxy) [![Build Status](https://travis-ci.org/producthunt/duxy.svg)](https://travis-ci.org/producthunt/duxy)
+
+RESTful resources.
+
+## Table of Contents
+
+  1. [Installation](#installation)
```
Comment (andreasklinger) [inline README.md:9]: show how to use it with redux in the early parts of the docs

File: README.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+      1. [get](#getname--path-)
+      1. [post](#postname--path-)
+      1. [put](putname--path-)
+      1. [patch](#patchname--path-)
+      1. [del](#delname--path-)
+      1. [namespace](#namespacename--path--fn)
+      1. [resources](#resourcesname--path-only-findone-findall-create-update-delete-fn)
+      1. [resource](#resourcename--path-only-findone-create-update-delete--fn)
+    1. [Adapter](#adapter)
+  1. [Development](#development)
```
Comment (andreasklinger) [inline README.md:23]: maybe show case a larger duxy file - eg one of producthunt so that people understand how it's awesome

**Review by andreasklinger — APPROVED** [review]

**Comment by andreasklinger** [comment]
lgtm

File: README.md (hunk 3 above)
Comment (RStankov) [reply README.md]: No linter for docs :P

**Pull request merged**